antidox/perspective.py

```diff
@@ -170,71 +158,101 @@ def main(argv):
   parser.add_argument('--csv_file', help='choose CSV file to process')
   parser.add_argument('--wiki_pagename', help='insert the talk page name')
   parser.add_argument('--content', help='specify a column in dataset to retreive data from')
-  parser.add_argument('--output', help='path for output file')
-  parser.add_argument('--suffix', help='output file suffix')
-  parser.add_argument('--project', help='project id for bigquery table')
-  args = parser.parse_args(argv)
+  parser.add_argument('--output', help='path for output file in cloud bucket')
+  parser.add_argument('--nd_output', help='gcs path to store ndjson results')
+  parser.add_argument('--project', help='project id for bigquery table', \
+                                   default='wikidetox-viz')
+  parser.add_argument('--gproject', help='gcp project id')
+  parser.add_argument('--temp_location', help='cloud storage path for temp files \
+                                              must begin with gs://')
+  args, pipe_args = parser.parse_known_args(argv)
   apikey_data, perspective, dlp = get_client()
-
-
-if __name__ == '__main__':
-  main(sys.argv[1:])
-  with beam.Pipeline(options=PipelineOptions()) as pipeline:
+  options = PipelineOptions(pipe_args)
+  gcloud_options = options.view_as(GoogleCloudOptions)
+  gcloud_options.project = 'google.com:new-project-242016'
+  gcloud_options.staging_location = 'gs://tj_cloud_bucket/stage'
+  gcloud_options.temp_location = 'gs://tj_cloud_bucket/tmp'
+  options.view_as(StandardOptions).runner = 'dataflow'
+  options.view_as(WorkerOptions).num_workers = 69
+  options.view_as(SetupOptions).save_main_session = True
+  with beam.Pipeline(options=options) as pipeline:
     if args.wiki_pagename:
       wiki_response = get_wikipage(args.wiki_pagename)
       wikitext = wiki_clean(wiki_response)
       text = wikitext.split("\n")
       comments = pipeline | beam.Create(text)
     if args.csv_file:
-      comments = pipeline | 'ReadMyFile' >> beam.io.ReadFromText(args.csv_file)
+      comments = pipeline | 'ReadMyFile' >> beam.io.ReadFromText(pd.read_csv(args.csv_file))
     if args.sql_query:
       comments = (
           pipeline
           | 'QueryTable' >> beam.io.Read(beam.io.BigQuerySource(
               query=args.sql_query,
-              use_standard_sql=True))
+              use_standard_sql=True)) \
           | beam.Map(lambda elem: elem[args.content]))
+
     # pylint: disable=fixme, too-few-public-methods
+    class NDjson(beam.DoFn):
+      """class for NDJson"""
+
+      # pylint: disable=fixme, no-self-use
+      # pylint: disable=fixme, inconsistent-return-statements
+      def process(self, element):
+        """Takes toxicity and dlp results and converst them to NDjson"""
+        try:
+          dlp_response = dlp_request(dlp, apikey_data, element)
+          perspective_response = perspective_request(perspective, element)
+          if contains_toxicity(perspective_response):
```

Comment:
With this logic if a comment contains both toxicity and pii, it looks like we will only write JSON with the 'Toxicity' element, and no 'pii_detected' element.  Would it be possible to rewrite this a little to be more like this:
```
data = {
  'comment': element,
  'toxicity': str(perspective_response['attributeScores']['TOXICITY']['summaryScore']['value'])},
  'pii_detected': str(pii_type)
}
```
And then have only 1 return statement?